tests/MongoDB.Driver.TestHelpers/DriverTestConfiguration.cs

```diff
@@ -124,6 +124,7 @@ public static DisposableMongoClient CreateDisposableClient(Action<ClusterBuilder
                 ? CoreTestConfiguration.ConnectionStringWithMultipleShardRouters.ToString()
                 : CoreTestConfiguration.ConnectionString.ToString();
             var clientSettings = MongoClientSettings.FromUrl(new MongoUrl(connectionString));
+            clientSettings.HeartbeatInterval = TimeSpan.FromMilliseconds(5);
```

Comment:
Spec with discussion: https://github.com/mongodb/specifications/pull/820

Reply:
yep, in these tests, we set a failpoint to actually fail read and write operations :) . Then the server type will be changed to Unknown. Then the retryable write/read will try to select a new server. In the old protocol, the server selection will force an immediate check. But in the new protocol, it won’t. We will just wait for the streamable ismaster to return and that will take 10 seconds. So it means that every test takes at least 10 seconds with the new protocol, every retryable test.

You may see in the above PR, that this change will be added for all drivers.